assemble/Advection_Diffusion_DG.F90

```diff
@@ -819,18 +838,114 @@ subroutine construct_advection_diffusion_dg(big_m, rhs, field_name,&
     call allocate(U_nl, U_nl_backup%dim, U_nl_backup%mesh, "LocalNonlinearVelocity")
     call set(U_nl, U_nl_backup)
 
-
-    Diffusivity=extract_tensor_field(state, trim(field_name)//"Diffusivity"&
+    Diffusivity_ptr => extract_tensor_field(state, trim(field_name)//"Diffusivity"&
          &, stat=stat)
     if (stat/=0) then
-       call allocate(Diffusivity, T%mesh, trim(field_name)//"Diffusivity",&
+       call allocate(Diffusivity, T%mesh, "Diffusivity",&
             FIELD_TYPE_CONSTANT)
        call zero(Diffusivity)
        include_diffusion=.false.
     else
-       ! Grab an extra reference to cause the deallocate below to be safe.
-       call incref(Diffusivity)
+       call allocate(Diffusivity, T%mesh, "Diffusivity")
+       call set(Diffusivity, Diffusivity_ptr)
+
        include_diffusion=.true.
+
+       if (have_option(trim(T%option_path)//"/prognostic"//&
+         &"/subgridscale_parameterisation::LES")) then
+
+         scalar_eddy_visc => extract_scalar_field(state, "DGLESScalarEddyViscosity", stat=stat)
+         call get_option(trim(T%option_path)//"/prognostic"//&
+              &"/subgridscale_parameterisation::LES/PrandtlNumber", prandtl, default=1.0)
+
+         ! possibly anisotropic eddy viscosity if using Ri dependency
+         call allocate(eddy_visc, mesh_dim(T), scalar_eddy_visc%mesh, &
+              & "EddyViscosity")
+         do i = 1, mesh_dim(T)
+           call set(eddy_visc, i, scalar_eddy_visc)
+         end do
+
+         ! apply Richardson dependence
+         if (have_option(trim(T%option_path)//"/prognostic"//&
+              &"/subgridscale_parameterisation::LES/Ri_c")) then
+
+           ewrite(2,*) 'Calculating Ri dependent eddy viscosity'
```

Comment:
This stuff is all way too long, it uses its own local variables; This really belongs in a separate subroutine.